app/app.js

```diff
@@ -2,12 +2,30 @@ const express = require("express");
 const app = express();
 const { getTopics } = require("./controllers/topics.controllers");
 const { getEndpoints } = require("./controllers/endpoints.controllers");
+const { getArticleById } = require("./controllers/articles.controllers");
 
 app.get("/api", getEndpoints);
+
 app.get("/api/topics", getTopics);
 
+app.get("/api/articles/:article_id", getArticleById);
+
 app.all("/api/*", (req, res, next) => {
-  res.status(404).send({ msg: "invalid path" });
+  res.status(404).send({ msg: "Invalid path" });
+});
+
+app.use((err, req, res, next) => {
+  if (err.msg && err.status) {
+    res.status(err.status).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
+});
+
+app.use((err, req, res, next) => {
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Bad request" });
+  }
```

Comment:
These error handlers are great! I'd recommend including a 500 error handler soon that will console log errors if they are getting past your other error handlers, this can be really useful for bug fixing as you develop 🐛